### PR TITLE
ca-certificates: build an `all` bottle

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This is needed for users on older versions of macOS who might not be
able to download the CA cert from curl.se.

### CODE REVIEW

File: Formula/ca-certificates.rb [hunk 1]
```diff
@@ -17,6 +17,8 @@ class CaCertificates < Formula
 
   def install
     pkgshare.install "cacert-#{version}.pem" => "cacert.pem"
+    # Ensure we have identical bottles by making the timestamps uniform across OS versions
+    touch pkgshare/"cacert.pem", mtime: path.mtime
```
Comment (carlocab) [inline Formula/ca-certificates.rb:21]: This isn't working for some reason. Weird.

**Comment by carlocab** [comment]
Hmm, no, this won't work. The problem seems to be in `brew`.

File: Formula/ca-certificates.rb (hunk 1 above)
Comment (carlocab) [reply Formula/ca-certificates.rb:21]: So this is changing the `mtime` of `cacert.pem` as expected. Except `brew` is changing it back upon bottling.

**Comment by carlocab** [comment]
Ah. No, I know what the problem is.

**Comment by carlocab** [comment]
So the problem is that we pass `--remote-time` to `curl` -- this works fine when downloading the source from https://curl.se/ca/cacert-2021-09-30.pem, but doesn't on Mojave, which falls back to formula-patches.

GitHub just sets `--remote-time` to the current time, which explains the timestamp diff between bottles.

**Comment by carlocab** [comment]
Dirty hack:

Make this `depends_on macos: [:catalina, :build]`. Older macOS versions won't be able to "build" this formula, but they won't need to because they can use the bottle.

**Comment by Bo98** [comment]
We can drop the mirror here and it should be good.

**Comment by carlocab** [comment]
That's unexpected:
```
==> brew audit ca-certificates --online --git --skip-style
==> FAILED
Error: undefined method `to_str' for false:FalseClass
Did you mean?  to_s
/opt/homebrew/Library/Homebrew/system_command.rb:176:in `block in expanded_args'
/opt/homebrew/Library/Homebrew/system_command.rb:170:in `map'
/opt/homebrew/Library/Homebrew/system_command.rb:170:in `expanded_args'
/opt/homebrew/Library/Homebrew/system_command.rb:124:in `command'
/opt/homebrew/Library/Homebrew/system_command.rb:186:in `each_output_line'
/opt/homebrew/Library/Homebrew/system_command.rb:53:in `run!'
/opt/homebrew/Library/Homebrew/system_command.rb:40:in `run'
/opt/homebrew/Library/Homebrew/system_command.rb:28:in `system_command'
/opt/homebrew/Library/Homebrew/utils/curl.rb:103:in `curl_with_workarounds'
/opt/homebrew/Library/Homebrew/utils/curl.rb:181:in `curl_output'
/opt/homebrew/Library/Homebrew/utils/curl.rb:309:in `curl_http_content_headers_and_checksum'
/opt/homebrew/Library/Homebrew/utils/curl.rb:230:in `block in curl_check_http_content'
/opt/homebrew/Library/Homebrew/utils/curl.rb:228:in `each'
/opt/homebrew/Library/Homebrew/utils/curl.rb:228:in `curl_check_http_content'
/opt/homebrew/Library/Homebrew/resource_auditor.rb:118:in `block in audit_urls'
/opt/homebrew/Library/Homebrew/resource_auditor.rb:109:in `each'
/opt/homebrew/Library/Homebrew/resource_auditor.rb:109:in `audit_urls'
/opt/homebrew/Library/Homebrew/resource_auditor.rb:38:in `block in audit'
/opt/homebrew/Library/Homebrew/resource_auditor.rb:33:in `each'
/opt/homebrew/Library/Homebrew/resource_auditor.rb:33:in `audit'
/opt/homebrew/Library/Homebrew/formula_auditor.rb:547:in `block in audit_specs'
/opt/homebrew/Library/Homebrew/formula_auditor.rb:533:in `each'
/opt/homebrew/Library/Homebrew/formula_auditor.rb:533:in `audit_specs'
/opt/homebrew/Library/Homebrew/formula_auditor.rb:798:in `block in audit'
/opt/homebrew/Library/Homebrew/formula_auditor.rb:793:in `each'
/opt/homebrew/Library/Homebrew/formula_auditor.rb:793:in `audit'
/opt/homebrew/Library/Homebrew/dev-cmd/audit.rb:186:in `block in audit'
/opt/homebrew/Library/Homebrew/dev-cmd/audit.rb:169:in `map'
/opt/homebrew/Library/Homebrew/dev-cmd/audit.rb:169:in `audit'
/opt/homebrew/Library/Homebrew/brew.rb:129:in `<main>'
```

**Comment by Bo98** [comment]
huh, ok. I'll have a look.

**Comment by Bo98** [comment]
https://github.com/Homebrew/brew/pull/12177

**Comment by carlocab** [comment]
Not sure if you want to wait for Homebrew/brew#12177 to re-run this, but I think this is ready to go. I'm logging off now though, so feel free to ship this when you think it's done.

**Comment by Bo98** [comment]
There'll be a free slot in the CI in a few minutes so might be worth trying to see if the fix worked.

**Review by Bo98 — APPROVED** [review]

**Pull request closed without merging**